src/ClientData/ThreadTrackDataProvider.cpp

```diff
@@ -47,6 +52,40 @@ void ThreadTrackDataProvider::OnCaptureComplete() {
        thread_track_data_manager_->GetAllScopeTreeTimerData()) {
     scope_tree_timer_data->OnCaptureComplete();
   }
+  UpdateTimerDurations();
+}
+
+[[nodiscard]] const std::vector<uint64_t>*
```

Comment:
No need for the [[nodiscard]] here, as it is already in the header.

Reply:
But I like it here. Is this a guideline?